src/lib/containers/menu/ComplexMenu.tsx

```diff
@@ -0,0 +1,52 @@
+import React from 'react'
+import { Box, IconButton, Tooltip } from '@mui/material'
+import IconSvg, { Icon } from '../IconSvg'
+import { DropdownMenu, DropdownMenuProps } from './DropdownMenu'
+
+/**
+ * Configuration for an IconButton inside the ComplexMenu
+ */
+export type IconButtonConfiguration = {
+  icon: Icon
+  tooltipText: string
+  onClick: (e: React.MouseEvent) => void
+}
+
+export type ComplexMenuProps = {
+  /*
+   * Configuration for IconButtons. Each entry corresponds to one button.
+   * See the IconButtonConfiguration type for more info.
+   */
+  iconButtons?: IconButtonConfiguration[]
+  /*
+   * Configuration for DropdownMenus. Each entry corresponds to one dropdown menu button, which
+   * itself can contain multiple groups of items.
+   * See the DropdownMenuProps type for more info.
+   */
+  dropdownMenus?: DropdownMenuProps[]
+}
+
+/**
+ * The ComplexMenu component allows you to create a generic menu with
+ * icon buttons and multiple dropdown menus that contain groups of items.
+ */
+export function ComplexMenu(props: ComplexMenuProps) {
```

Comment:
The one (and probably only) complex menu is the SWC Entity tools menu. This seemed like a logical place to divide business logic from plain UI logic